src/pybind/mgr/dashboard/frontend/src/app/shared/components/icon/icon.component.ts

```diff
@@ -0,0 +1,15 @@
+import { Component, Input } from '@angular/core';
+import { ICON_TYPE, Icons, IconSize } from '../../enum/icons.enum';
+
+@Component({
+  selector: 'cd-icon',
+  templateUrl: './icon.component.html',
+  styleUrl: './icon.component.scss'
+})
+export class IconComponent {
+  @Input() type!: keyof typeof ICON_TYPE;
```

Comment:
looks like this icon component only accept types and they are like the defined types? But I'd assume this will accept the class names too, so this can be used for any generic icons?

Reply:
yup, i was mostly concerned with html not being generic but now it looks good and I understand it better. Thanks @afreen23